### ISSUE TITLE
SVG shapes add vertex at origin on Chrome.

### ISSUE THREAD

**Issue by GKFX** [processing-js/processing-js#137]
Quote from http://stackoverflow.com/questions/16415304/how-to-correctly-cleanly-draw-svg-graphics-using-processingjs/28285366#28285366:
I'm getting started with ProcessingJS and I'm currently playing with SVG a bit. Unfortunately I've ran into a strange behaviour displaying SVG. Here is the output:

![74poq](https://cloud.githubusercontent.com/assets/5357642/6007544/f238f324-ab12-11e4-8885-4e0e4a277bdd.png)

and here is the code that produces that:

````html
<script src="http://ajax.googleapis.com/ajax/libs/jquery/2.0.0/jquery.min.js"></script>
<script src="http://cloud.github.com/downloads/processing-js/processing-js/processing-1.4.1.min.js"></script>
<script>
$(document).ready(function() {  
    $('body').append($('<canvas id="preview"><p>Your browser does not support the canvas tag.</p></canvas>'));
    function onPJS(p) {
        var glasses;
        p.setup = function(){
            glasses = p.loadShape("brighteyes.svg");
            p.size(Math.floor(glasses.width), Math.floor(glasses.height)+Math.floor(glasses.height * .25));
            //p.shape(glasses,0,0);
            p.frameRate(1);
        }
        //*
        p.draw = function() {
            p.background(32);
            p.shape(glasses, 0, 0);
            console.log(p.mousePressed);//prints undefined
        };
        //*/
    }
    new Processing(document.getElementById("preview"), onPJS);  
});
</script>
````

I'm experiencing this odd rendering (renderer seems to place a vertex at 0,0 for the shape) on OSX 10.8 on Chrome Version 26.0.1410.65 (but not on Safari (6.0 (8536.25))). You can run the code here.

How do I get read of these weird rendering bug ?

There is another unexpected thing happening: mousePressed prints undefined, but might address that in a different question.

**Comment by GoToLoop** [comment]
Dunno much about those bugs but I've got some points:  
* PJS framework works best in "JavaScript Mode" w/ Java syntax.
* That is so b/c there are some gotchas which PJS' transpiler needs to deal w/.
* Besides auto-generating a basic web-exporter of the sketch for us.
* 1 of those gotchas is _mousePressed_ you got there.
* In Java functions can have the same name as a variable. They got their own domain!
* While in JS functions are stored in variables too. So functions can't share labels w/ other types!
* In PJS, it is function **mousePressed()**. Its corresponding `boolean` is *__mousePressed*! @_@
* You can check that out here: http://processingjs.org/tools/processing-helper.html
* Type this code below and click @ "Convert to JS" button and see their differences:
```Processing
void mousePressed() {
  if (mousePressed);
}
```
About your main problem perhaps it lacks preloading directives?
`/** @pjs preload = "brighteyes.svg"; */`

http://processingjs.org/reference/pjs%20directive/
http://processingjs.org/reference/preload/

Or at least you should load that in your HTML file itself!? O_o

**Comment by Pomax** [comment]
this looks like a bug in the SVG pathing code. Can you reduce that SVG to the smallest file that that still has that problem? If so, we can use that for debugging where things go wrong.

**Comment by GKFX** [comment]
I cannot reproduce the bug on my own machine (Ubuntu Linux 14.04, latest Firefox/Chrome/Chromium), and I'm struggling to get hold of the old copy of Chrome for Linux. Could you go to [the user who had the fault](http://stackoverflow.com/questions/16415304/how-to-correctly-cleanly-draw-svg-graphics-using-processingjs/28285366#28285366) and ask them directly for a minimal case? Alternatively, just use the full version at http://lifesine.eu/so/brighteyes.svg.

**Comment by Pomax** [comment]
In terms of the SVG, I'm going to call this resolved due to browsers getting bugs fixed. I can't reproduce their malrendering on either FF or Chrome on Windows and OSX.

The note about not overloading variables and function names is valid, though (note that you **can** overload functions with distinct inputs, you just can't use the same name for a function *and* a variable)
